apps/emqx_bridge/src/emqx_bridge_api.erl

```diff
@@ -390,40 +390,50 @@ schema("/nodes/:node/bridges/:id/operation/:operation") ->
     Conf = filter_out_request_body(Conf0),
-    case emqx_bridge:lookup(BridgeType, BridgeName) of
-        {ok, _} ->
-            {400, error_msg('ALREADY_EXISTS', <<"bridge already exists">>)};
-        {error, not_found} ->
-            case ensure_bridge_created(BridgeType, BridgeName, Conf) of
-                ok -> lookup_from_all_nodes(BridgeType, BridgeName, 201);
-                {error, Error} -> {400, Error}
-            end
+    case has_redact_value(Conf) of
+        false ->
+            case emqx_bridge:lookup(BridgeType, BridgeName) of
+                {ok, _} ->
+                    {400, error_msg('ALREADY_EXISTS', <<"bridge already exists">>)};
+                {error, not_found} ->
+                    case ensure_bridge_created(BridgeType, BridgeName, Conf) of
+                        ok -> lookup_from_all_nodes(BridgeType, BridgeName, 201);
+                        {error, Error} -> {400, Error}
+                    end
+            end;
+        _ ->
+            {400, error_msg('BAD_REQUEST', <<"There is obfuscated sensitive data">>)}
     end;
 '/bridges'(get, _Params) ->
     {200,
         zip_bridges([
-            [format_resp(Data, Node) || Data <- emqx_bridge_proto_v1:list_bridges(Node)]
+            [redact(format_resp(Data, Node)) || Data <- emqx_bridge_proto_v1:list_bridges(Node)]
          || Node <- mria_mnesia:running_nodes()
         ])}.
 
 '/bridges/:id'(get, #{bindings := #{id := Id}}) ->
     ?TRY_PARSE_ID(Id, lookup_from_all_nodes(BridgeType, BridgeName, 200));
 '/bridges/:id'(put, #{bindings := #{id := Id}, body := Conf0}) ->
     Conf = filter_out_request_body(Conf0),
-    ?TRY_PARSE_ID(
-        Id,
-        case emqx_bridge:lookup(BridgeType, BridgeName) of
-            {ok, _} ->
-                case ensure_bridge_created(BridgeType, BridgeName, Conf) of
-                    ok ->
-                        lookup_from_all_nodes(BridgeType, BridgeName, 200);
-                    {error, Error} ->
-                        {400, Error}
-                end;
-            {error, not_found} ->
-                {404, error_msg('NOT_FOUND', <<"bridge not found">>)}
-        end
-    );
+    case has_redact_value(Conf) of
+        false ->
+            ?TRY_PARSE_ID(
+                Id,
+                case emqx_bridge:lookup(BridgeType, BridgeName) of
+                    {ok, _} ->
+                        case ensure_bridge_created(BridgeType, BridgeName, Conf) of
+                            ok ->
+                                lookup_from_all_nodes(BridgeType, BridgeName, 200);
+                            {error, Error} ->
+                                {400, Error}
+                        end;
+                    {error, not_found} ->
+                        {404, error_msg('NOT_FOUND', <<"bridge not found">>)}
+                end
+            );
+        _ ->
+            {400, error_msg('BAD_REQUEST', <<"There is obfuscated sensitive data">>)}
```

Comment:
if this PUT is used for creation, then it's reasonable to return 400 if ****** is provided as password.

AFAIK, PUTs are used for updating, in this caes the handler should replace it with the currently configured password.

Reply:
There is and only POST will create a bridge, so there is no need to handle creating here